pandas/core/algorithms.py

```diff
@@ -1315,6 +1315,11 @@ def take_nd(arr, indexer, axis=0, out=None, fill_value=np.nan, mask_info=None,
         undefined if allow_fill == False and -1 is present in indexer.
     """
 
+    if is_sparse(arr):
```

Comment:
Hmm, this seems sub-optimal. We shouldn't have to densify just to slice, right?

Can you take a look at `SparseAray.take` to see if any of that logic can be refactored / reused?

And since you're touching performance-related code, could you run the sparse-related ASVs?

Reply:
thank you for pointing this out. exactly what I was looking for. Will run the ASVs